### PR TITLE
rejig to make external device lif control better for tdma

### PR DESCRIPTION
**Pull request by alan-stokes**
a rejig

https://github.com/SpiNNakerManchester/sPyNNaker8/pull/432 jenkins

### CODE REVIEW

File: spynnaker/pyNN/external_devices_models/external_device_lif_control.py [hunk 1]
```diff
@@ -20,12 +20,13 @@
 from spynnaker.pyNN.models.defaults import default_initial_values,\
     default_parameters
 from spynnaker.pyNN.models.neuron.input_types import InputTypeCurrent
-from spynnaker.pyNN.models.neuron.neuron_models import (
-    NeuronModelLeakyIntegrateAndFire)
 from spynnaker.pyNN.models.neuron.synapse_types import SynapseTypeExponential
 from .external_device_lif_control_vertex import ExternalDeviceLifControlVertex
 from .threshold_type_multicast_device_control import (
     ThresholdTypeMulticastDeviceControl)
+from ..models.neuron.neuron_models.\
```
Comment (rowleya) [inline spynnaker/pyNN/external_devices_models/external_device_lif_control.py:27]: 1) Don't use relative imports like this.  Only import from the current folder, or else go for the full thing.
2) Why do you need this?  You are using standard LIF in the C code...

File: spynnaker/pyNN/models/neuron/neuron_models/neuron_model_external_device_control.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
+# Copyright (c) 2017-2019 The University of Manchester
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/neuron_models/neuron_model_external_device_control.py:1]: I'm confused about the need for this entire file.  It looks like a clone of LIF.  Why?

File: neural_modelling/src/neuron/implementations/neuron_impl_external_devices.h [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+                additional_inputs, soma_voltage);
+
+        // update neuron parameters
+        state_t result = neuron_model_state_update(
+                NUM_EXCITATORY_RECEPTORS, exc_input_values,
+                NUM_INHIBITORY_RECEPTORS, inh_input_values,
+                external_bias, this_neuron);
+
+        // determine if a spike should occur
+        has_spiked = threshold_type_is_above_threshold(the_threshold_type);
```
Comment (rowleya) [inline neural_modelling/src/neuron/implementations/neuron_impl_external_devices.h:427]: I suspect this is misnamed; it isn't really a threshold type any more, and it isn't a spike.

Comment (alan-stokes) [reply neural_modelling/src/neuron/implementations/neuron_impl_external_devices.h:427]: inherited from before, but will change.

File: spynnaker/pyNN/external_devices_models/external_device_lif_control.py (hunk 1 above)
Comment (alan-stokes) [reply spynnaker/pyNN/external_devices_models/external_device_lif_control.py:27]: coz i couldnt get the compiler to play ball without it.

Comment (rowleya) [reply spynnaker/pyNN/external_devices_models/external_device_lif_control.py:27]: The python compiler?  You  should be able to use just the LIF model I believe

File: spynnaker/pyNN/models/neuron/neuron_models/neuron_model_external_device_control.py (hunk 2 above)
Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/neuron_models/neuron_model_external_device_control.py:1]: was when thinking a different route. killing

File: spynnaker/pyNN/external_devices_models/external_device_lif_control.py (hunk 1 above)
Comment (alan-stokes) [reply spynnaker/pyNN/external_devices_models/external_device_lif_control.py:27]: 1. was pycharm doing pycharmness. am correcting when replacing with old lif

Comment (alan-stokes) [reply spynnaker/pyNN/external_devices_models/external_device_lif_control.py]: fixed

File: neural_modelling/src/neuron/implementations/neuron_impl_external_devices.h (hunk 3 above)
Comment (alan-stokes) [reply neural_modelling/src/neuron/implementations/neuron_impl_external_devices.h]: fixed

**Pull request merged**